### DISCUSSION TITLE
TreeView - Can not put LabelGroup in expandable TreeViewItem due to nested button elements

### DISCUSSION THREAD

**Discussion by ianw** [patternfly/patternfly#5488] [Feature requests]
I have made a ```TreeView``` with ```item.name``` as a more complex object, where I wanted to use a ```LabelGroup``` within it; but this leads to 

```
Warning: validateDOMNesting(...): <button> cannot appear as a descendant of <button>.
```

A screenshot is below

![image](https://user-images.githubusercontent.com/108448/193949265-155056c8-6c18-4aae-93a8-98c6b09353ec.png)

I guess the problem here is that the expandable-row (```pf-c-tree-view__node``` above) from the ```TreeView``` is made with a ```button``` (highlighed above), but then the ```LabelGroup``` wants to use a button to have the group expansion too; e.g.

![image](https://user-images.githubusercontent.com/108448/193949548-5ad55be2-4652-4eba-8797-f68d2f736e2d.png)

A suggestion might be that the ```TreeView``` should somehow be implemented with a ```div```/```onclick``` instead of a ```button``` to give a bit more flexibility to the internal tree view elements?

**Comment by mcoker** [comment]
cc @mcarrano @mmenestr wdyt?

**Comment by mcarrano** [comment]
@ianw can you tell us more about the underlying use case and why you are wanting to put a label group in the tree?
